app/models/notification.rb

```diff
@@ -0,0 +1,48 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+
+  belongs_to :user
+  belongs_to :course
+  def self.center_popup(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    set_options(notification, options)
+    notification.n_type = 'center_popup'
+    notification.save
+  end
+
+  def self.right_side_popup(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    set_options(notification, options)
+    notification.n_type = 'right_side_popup'
+    notification.save
+  end
+
+  def self.activity_only(target_course_id, target_user_id, options = {})
+    notification = Notification.new
+    notification.course_id = target_course_id
+    notification.user_id = target_user_id
+    notification.action = options[:action]
+    notification.activity_content = options[:activity_content]
+    notification.link = options[:link]
+    notification.save
+  end
+
+  def self.set_options(notification, options)
```

Comment:
hum...It will, and that will be an empty feed...That means I should avoid this case?